src/thing_creature.c

```diff
@@ -1658,19 +1659,26 @@ void creature_cast_spell(struct Thing *castng, long spl_idx, long shot_lvl, long
     // Check if the spell can be self-casted
     if (spinfo->caster_affected)
     {
-        i = (long)spinfo->caster_affect_sound;
-        if (i > 0)
-          thing_play_sample(castng, i, NORMAL_PITCH, 0, 3, 0, 4, FULL_LOUDNESS);
+        if (spinfo->caster_affect_sound > 0)
+          thing_play_sample(castng, spinfo->caster_affect_sound, NORMAL_PITCH, 0, 3, 0, 4, FULL_LOUDNESS);
         apply_spell_effect_to_thing(castng, spl_idx, cctrl->explevel);
     }
     // Check if the spell has an effect associated
     if (spinfo->cast_effect_model != 0)
     {
-        struct Thing* efthing = create_effect(&castng->mappos, spinfo->cast_effect_model, castng->owner);
+        struct Thing* efthing;
+        if (spinfo->cast_effect_model > 0)
+        {
+            efthing = create_effect(&castng->mappos, spinfo->cast_effect_model, castng->owner);
+        }
+        else
+        {
+            efthing = create_effect_element(&castng->mappos, ~(spinfo->cast_effect_model)+1, castng->owner);
+        }
```

Comment:
idk if we have many of these if's but maybe we could have a create_effect_or_effect_element in 1 function that handles both